### PR TITLE
refactor: s3manager to use awsutils for creating session

### PR DESCRIPTION
**Pull request by koladilip**
# Description
Refactor S3 Manager GetConfig to use mapstructure.
Refactor S3 Manager getSession to use awsutils.
Avoid using session.Must as it will panic.
Add role supoort for S3 destination.

## Notion Ticket
[FTR for Streaming Destinations](https://www.notion.so/rudderstacks/AWS-FTR-for-streaming-destination-78b63869ac904ba49b0018a527ccb343)

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Comment by koladilip** [comment]
I have tested successfully role based access for S3 destination and it is backward compatible.

File: utils/awsutils/session.go [hunk 1]
```diff
@@ -9,70 +9,98 @@ import (
...
+	Timeout          time.Duration `mapstructure:"timeout"`
 }
 
 func createRoleSessionName(serviceName string) string {
 	return fmt.Sprintf("rudderstack-aws-%s-access", strings.ToLower(serviceName))
 }
 
-func createDefaultSession(config *SessionConfig) *session.Session {
-	return session.Must(session.NewSession(&aws.Config{
+func createDefaultSession(config *SessionConfig) (*session.Session, error) {
```
Comment (saikumarrs) [inline utils/awsutils/session.go:36]: No value for `error` is returned from the function.

Comment (koladilip) [reply utils/awsutils/session.go:36]: session.NewSession returns (response, error) so I am just passing that.

**Review by psrikanth88 — APPROVED** [review]
LGTM

**Pull request merged**